### PR TITLE
Warn if test was not executed

### LINKED ISSUE #40: Tests that are not actually run are marked as passing
When I trigger an individual test in a testify suite, it is marked as passing even if no test was run.  The run spec produces the following command, which doesn't match the test, due to it being in a testify suite and needing `^SuiteName/TestName$`

```
  command = { "go", "test", "-json", "-v", "-race", "-count=1", "-timeout=60s", "-run", "^TestName$" },
```

 I know that's not supported yet, but if you try to run an individual test and no tests are run, it seems like it should fail and warn, not pass silently

### PR DESCRIPTION
**Pull request by fredrikaverpil**
This can happen if you e.g. are using testify. Fixes #40.

### Demo

https://github.com/fredrikaverpil/neotest-golang/assets/994357/7d17e740-8594-4595-a327-6ca9d3d4352e

### CODE REVIEW

**Review by jstensland — APPROVED** [review]
tested and resolves https://github.com/fredrikaverpil/neotest-golang/issues/40 for me

**Comment by fredrikaverpil** [comment]
@jstensland thanks for testing (and thanks for reporting!) I'll merge this shortly together with some other minor improvements.

File: lua/neotest-golang/results_test.lua [hunk 1]
```diff
@@ -75,17 +80,29 @@ function M.results(spec, result, tree)
...
+    -- assign status code, as long as the test was found
+    if result.code == 0 then
+      results[spec.context.id].status = "passed"
+    else
+      results[spec.context.id].status = "failed"
+    end
+  else
+    if options.get().warn_test_not_executed == true then
+      vim.notify(
+        "Could not find test: " .. convert.to_gotest_test_name(spec.context.id),
```
Comment (fredrikaverpil) [inline lua/neotest-golang/results_test.lua:93]:
```suggestion
        "Test not executed: " .. convert.to_gotest_test_name(spec.context.id),
```

File: lua/neotest-golang/results_test.lua [hunk 2]
```diff
@@ -45,10 +43,17 @@ function M.results(spec, result, tree)
     if line.Action == "output" and line.Output ~= nil then
       -- record output, prints to output panel
       table.insert(test_result, line.Output)
+
+      -- if test was not run, mark it as skipped
+
+      -- if line contains "no test files" or "no tests to run", mark as skipped
+      if string.match(line.Output, "no tests to run") then
+        test_not_found = true
```
Comment (fredrikaverpil) [inline lua/neotest-golang/results_test.lua:51]:
```suggestion
        test_not_executed = true
```

File: lua/neotest-golang/results_test.lua (hunk 1 above)
Comment (fredrikaverpil) [inline lua/neotest-golang/results_test.lua:93]:
```suggestion
        "Could not execute test: " .. convert.to_gotest_test_name(spec.context.id),
```

**Pull request merged**